subsys/net/ip/net_context.c

```diff
@@ -1346,6 +1346,11 @@ NET_CONN_CB(tcp_syn_rcvd)
 
 		net_tcp_print_recv_info("SYN", pkt, NET_TCP_HDR(pkt)->src_port);
 
+		if (net_tcp_get_state(tcp) != NET_TCP_LISTEN) {
+			NET_DBG("we cannot process 2 SYNs here simultaneously.");
+			retrun NET_DROP;
```

Comment:
@jukkar   Multiple clients access at the same time ,the second client will received a RST after handshake. (According to a tester's report on previous version)